product_specifications/DVjtom_67/readme.md

```diff
@@ -0,0 +1,149 @@
+
+        Scenario: Recieving usage stats from the Tracker for non Datacite DOIs (transfer or malformed)
+            Given that Stats Processing Service received the analytics from the Tracker
+             When data is sent to the processor
+             And the DOI name is not in our index.
+             Then the processor must discard the data.
+             And it must return a 404 Error to the tracker.
+
+Feature: Acessing Usage Stats Processing
+
+        Scenario: Displaying Usage Stats from Tracker in the REST API
+            Given that usage stats exist for a DOI
+             When a user visits to the enpoints '/dois/{doi-name}'
+             Then the user should be able to see the usage stats counts in the response
+
+        Scenario: Displaying Usage Stats from Tracker in the GrapQL API
+            Given that usage stats exist for a DOI
+             When a user visits to the endpoint '/graphql' and makes a query for a viewsCount or downloadsCount
+             Then the user should be able to see the usage stats counts in the response
+
+        Scenario: Displaying Usage Stats from Tracker in the Data metrics badge
+            Given that usage stats exist for a DOI
+             When a user visits a page that include the Data metrics badge for that DOI
+             Then the user should be able to see the usage stats counts in the badge
+
+        Scenario: Displaying Usage Stats from Tracker in the EventData API
+            Given that usage stats exist for a DOI
+             When a user visits to the enpoints '/events/doi={doi-name}'
+             Then the user should be able to see the usage stats counts as events in the response
+
+        Scenario: Displaying Usage Stats from Tracker in Fabrica
+            Given that usage stats exist for a DOI
+             When a user visits to the Usage Tracking Dashboard Tab
+             Then the user should be able to see the usage stats in different views.
+
+        Scenario: Displaying Usage Stats from Tracker in SUSHI Reports
+            Given that usage stats exist for a Repository on a given month
+             When a user visits to the enpoint '/reports/{report-id}'
+             Then the user should be able to download a usage report (SUSHI JSON) 
+
+```
+
+
+## Non Functional Requirements
+
+- As an repository admin, I would like the collected usage stats were available, in the different interfaces, within 10 days after the end of month of collection.
+
+## Feature Flag
+
+This feature is implemented behind the `usage-processing-service` feature flag and disabled by default.
+Once we've confirmed the feature is deemed stable, we remove the feature flag to publish the feature as GA.
+
+## Technical Feasibility
+
+There are a few open questions about technical feasibility:
+
+- How can plausible be used display usage stats? 
+- How can relaiably process usage stats according to the CoP and produce similar stats as with the weblogs processing?
```

Comment:
I would like to see ongoing infrastructure cost estimates for this work.

Reply:
We can do some guess work but I think we will need a tech spec to be able to calculate that.

Reply:
Makes sense